kernel/safe_typing.ml

```diff
@@ -873,7 +873,10 @@ let export_private_constants eff senv =
   let senv = List.fold_left add_constant_aux senv bodies in
   exported, senv
 
-let add_constant l decl senv =
+let add_constant ?typing_flags l decl senv =
+  let senv = Option.cata
+      (fun typing_flags -> set_typing_flags typing_flags senv)
+      senv typing_flags in
```

Comment:
Shouldn't we reset the flags after adding the constant?

Reply:
Obviously yes ups